src/main/cpp/util/md5.cc

```diff
@@ -159,7 +159,8 @@ void Md5Digest::Finish(unsigned char digest[16]) {
 
   /* Put the 64-bit file length in *bits* at the end of the buffer.  */
   unsigned int size = (ctx_buffer_len < 56 ? 64 : 128);
-  uint32_t words[2] = { count[0] << 3, (count[1] << 3) | (count[0] >> 29) };
+  uint32_t words[2] = { htole32(count[0] << 3),
```

Comment:
This looks reasonable, AFAIU what it does is it puts the length measured in bits in little-endian format at the buffer; the code is basically "multiply a 64-bit number by 8".